pkg/apis/garden/validation/validation_shoot.go

```diff
@@ -1809,7 +796,10 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *garden.ShootSpec, deletionTimesta
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.GCP.Networks.VPC, oldSpec.Cloud.GCP.Networks.VPC, gcpPath.Child("networks", "vpc"))...)
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.GCP.Networks.Internal, oldSpec.Cloud.GCP.Networks.Internal, gcpPath.Child("networks", "internal"))...)
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.GCP.Networks.Workers, oldSpec.Cloud.GCP.Networks.Workers, gcpPath.Child("networks", "workers"))...)
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.GCP.Zones, oldSpec.Cloud.GCP.Zones, gcpPath.Child("zones"))...)
+
```

Comment:
azure zones are not validated in the first place, so "removing the restriction" is already done, also not sure if zones are used in the first place.

Reply:
@dkistner what's the reason that there is no validation? Is it easily possible to switch a zoned cluster to a non-zoned cluster or vice versa?

Reply:
ping  @dkistner

Reply:
There is some validation which forbid to move from zoned to non zoned https://github.com/gardener/gardener/blob/master/pkg/apis/garden/validation/validation.go#L1795-L1800.

I did not allow this because moving from non-zoned to zoned is not possible in all regions (zones are not available in all regions) and there would be also a need to migrate the load balancer from one sku to another, which would need to delete all LB services and recreate them.

Reply:
OK, thanks, makes total sense. @zanetworker please adapt this logic to allow adding new zones.

Reply:
Does it mean, after this pull request it is still impossible to switch the shoot from a non-zoned to a zoned one?

Reply:
I guess yes. This PR only targets adding additional zones to the already existing zones.

The scenario you described is not a goal of this PR. @dkistner are there any plans to enable this?

Reply:
No not yet, because this would come also with a lot of other efforts for instance additional validation if cluster is in a zone supporting region etc.